tests/samplers_tests/tpe_tests/test_parzen_estimator.py

```diff
@@ -322,3 +368,33 @@ def test_calculate(
     np.testing.assert_almost_equal(s_weights, expected["weights"])
     np.testing.assert_almost_equal(s_mus, expected["mus"])
     np.testing.assert_almost_equal(s_sigmas, expected["sigmas"])
+
+
+@pytest.mark.parametrize(
+    "weights",
+    [
+        lambda x: np.zeros(x),
+        lambda x: -np.ones(x),
+        lambda x: float("inf") * np.ones(x),
+        lambda x: -float("inf") * np.ones(x),
+        lambda x: np.asarray([float("nan") for _ in range(x)]),
+    ],
+)
+def test_invalid_weights(weights: Callable[[int], np.ndarray]) -> None:
+    parameters = _ParzenEstimatorParameters(
+        prior_weight=1.0,
+        consider_prior=False,
+        consider_magic_clip=False,
+        consider_endpoints=False,
+        weights=weights,
```

Comment:
[FYI] Although it would be bad to have multiple `inf` weights, one could in principle allow `prior_weight == inf` such that every sample is drawn from the prior distribution. 
However, if that adds code complexity, I think we might as well simply disallow it.